src/rt-app_taskgroups.c

```diff
@@ -157,23 +161,34 @@ static int cgroup_get_cpu_controller_mount_point(void)
 	}
 
 	while (ent = getmntent(mounts)) {
-		if (strcmp(ent->mnt_type, "cgroup"))
-			continue;
+		if (!strcmp(ent->mnt_type, "cgroup")) {
+			if (!hasmntopt(ent, "cpu"))
+				continue;
+
+			ctrl.mount_point = malloc(strlen(ent->mnt_dir) + 1);
+			ctrl.version = 1;
+			if (!ctrl.mount_point) {
+				perror("malloc");
+				break;
+			}
 
-		if (!hasmntopt(ent, "cpu"))
-			continue;
+			strcpy(ctrl.mount_point, ent->mnt_dir);
+			log_debug(PIN "cgroup cpu controller mountpoint [%s] found", ent->mnt_dir);
+			ret = 0;
+			break;
+		} else if (!strcmp(ent->mnt_type, "cgroup2")) {
+			ctrl.mount_point = malloc(strlen(ent->mnt_dir) + 1);
```

Comment:
I have seen system with both cgroup (for cpu) and cgroup2. I'm not sure that getmntent() returns a sorted listed of mount point which means that if you check for cgroup2 before cgroup you will return whereas the cpu controller is in cgroup. you need to check in cgroup.controllers that cpu controller is available